### ISSUE TITLE
Hide/show cursor ANSI escape codes get printed even without a tty

### ISSUE THREAD

**Issue by MartinHammarstedt** [rsalmei/alive-progress#55]
The hide/show cursor ANSI escape codes get printed even when piping the output to `less`, for example:

```
> python3 test.py | less
ESC[?25lESC[?25h|████████████████████████████████████████| 50/50 [100%] in 5.0s (9.98/s)
```

**Comment by rsalmei** [comment]
Hey @MartinHammarstedt,

Humm, the bar correctly didn't animate, but the final receipt included those escape codes, is that right?

**Comment by MartinHammarstedt** [comment]
Correct! And apparently they also show up on every `print()`:
```
ESC[?25lESC[2K^Mon 0: Hello
ESC[2K^Mon 1: Hello
ESC[2K^Mon 2: Hello
ESC[2K^Mon 3: Hello
ESC[2K^Mon 4: Hello
ESC[?25h|████████████████████████████████████████| 5/5 [100%] in 2.5s (2.00/s)
```

**Comment by rsalmei** [comment]
This is a different one, on every print there is one to clear the entire line.
When animating these are mandatory, but when piping maybe not... I assume these are also when piping, right?

I'm curious why they are showing up like that, are you on windows?

**Comment by MartinHammarstedt** [comment]
The above is when piping the output to `less`, on Ubuntu. Redirecting the output to a file instead leads to a similar result (how the escape codes are displayed differs depending on which editor I open the resulting file in).

I imagine no such codes are needed when piping, as then there's no progress bar to clear before printing.

**Comment by rsalmei** [comment]
Yeah, they are really not needed. Nice catch!

I'll fix that in the 2.0, hope you can wait.
I'm struggling very much with those wide chars you brought to my attention... Please read the UPDATE I wrote on #51 ...

**Comment by rsalmei** [comment]
Closing the ready issues to reduce the number of opened ones.
But fell free to contact me if anything arises!
